src/lang-promql/complete/hybrid.ts

```diff
@@ -93,29 +94,26 @@ export class HybridComplete implements CompleteStrategy {
     const { state, pos } = context;
     const tree = state.tree.resolve(pos, -1);
     if (tree.parent?.type.id === MetricIdentifier && tree.type.id === Identifier) {
+      let nonMetricCompletions = [autocompleteNode.functionIdentifier, autocompleteNode.aggregateOp, autocompleteNode.binOp];
+
+      if (tree.parent?.parent?.parent?.parent?.type.id === BinaryExpr) {
```

Comment:
yeah we really need a sort of Tree Walker. Looks like a bit too random and too dependent of lezer-promql.

Reply:
Yeah. It will always be dependent on lezer-promql though, even if we can abstract it a bit at some point. Basically we have to guess (or rather, experiment) what Lezer guesses what an expression is supposed to be when it's not a fully valid expression.